lib/helpers/launch_scripts.ex

```diff
@@ -0,0 +1,35 @@
+defmodule Burrito.Helpers.LauncScripts do
+  require Logger
+
+  def copy_launch_scripts(self_path, release_path, release_name) do
```

Comment:
This function is a little bit difficult for me to follow. What do you think about making it more declarative by defining the scripts upfront, and then iterating over them with a comprehension? I haven't tested it, but something like this:

```elixir
  def copy_launch_scripts(self_path, release_path, release_name) do
    src_dir = scripts_dir(self_path)
    scripts = launch_scripts(release_path, release_name)

    for {src_name, dest_dir, dest_name} <- scripts do
      src_path = Path.join(src_dir, src_name)
      dest_path = Path.join(dest_dir, dest_name)

      File.copy!(src_path, dest_path)
      File.chmod!(dest_path, 0o744)
    end
  end

  defp launch_scripts(release_path, release_name) do
    bins_dir = bins_dir(release_path)
    erts_dir = erts_dir(release_path)

    [
      {"posix_start.sh", bins_dir, release_name},
      {"win_start.bat", bins_dir, release_name <> ".bat"},
      {"posix_erl.sh", erts_dir, "erl"},
      {"posix_erts_start.sh", erts_dir, "start"}
    ]
  end

  defp scripts_dir(release_path) do
    Path.join(release_path, "src/scripts")
  end

  defp bins_dir(release_path) do
    Path.join(release_path, "bin")
  end

  defp erts_dir(release_path) do
    release_path
    |> Path.join("erts-*/bin")
    |> Path.wildcard()
    |> hd()
  end
```

The main benefit here is that it becomes clearer that we're performing fundamentally the same operations for each script, and that it's only the paths and filenames that differ.

There's also two small functional changes in my version:
1) We also `chmod!` the Windows script, which technically isn't necessary, but shouldn't impact anything
2) Instead of `List.first/1` I call `hd/1`, so that we'll fail with an `ArgumentError` if `ERTS` isn't found, rather than just silently returning `nil`

Reply:
Oh i like this structure a lot, I'll refactor!